.github/workflows/ci.yml

```diff
@@ -0,0 +1,51 @@
+name: CI
+
+on:
+  pull_request:
+    types:
+      - "opened"
+      - "synchronize"
+    paths-ignore:
+      - "README.md"
+      - ".gitignore"
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    env:
+      GOPATH: /home/runner/go
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+      with:
+        submodules: recursive
+    - name: Setup Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.19
+    - name: Format check
+      run: make fmt
+    - name: golangci-lint
+      uses: golangci/golangci-lint-action@v3
+      with:
+        version: v1.52
+    - name: Check generated code
+      run: make generate
+
+  test:
+    runs-on: ubuntu-latest
+    env:
+      GOPATH: /home/runner/go
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+      with:
+        submodules: recursive
+    - name: Setup Go
+      uses: actions/setup-go@v3
```

Comment:
May want to use v4 or v5 https://github.com/actions/setup-go

Reply:
updated to v5